File: backend/DeliveryService/dto/response/ApiResponse.py
from datetime import date
from typing import Optional, TypeVar, Generic

from pydantic import BaseModel

from dto.response.coordinatesResponse import CoordinatesResponse

T = TypeVar('T')


class ApiResponse(BaseModel, Generic[T]):
    code: int = 200
    isSuccess: bool = True
    result: Optional[T]
    message: str
    timestamp: date=date.today()

File: backend/logistics-payments-service/src/main/java/com/tdtu/logistics_payments_service/dto/response/ApiResponse.java
package com.tdtu.logistics_payments_service.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse<T> {
	@Builder.Default
	int code = 200;

	@Builder.Default
	@JsonProperty("isSuccess")
	boolean isSuccess = Boolean.TRUE;

	@JsonProperty("result")
	T result;

	@JsonProperty("message")
	String message;

	@JsonProperty("timestamp")
	LocalDate timestamp;
}
